### PR TITLE
Add Labels To Service Brokers

### PR DESCRIPTION
**Pull request by KirilKabakchiev**
Adds support for labels to brokers

### CODE REVIEW

File: api/broker/broker_controller.go [hunk 1]
```diff
@@ -227,6 +229,15 @@ func (c *Controller) patchBroker(r *web.Request) (*web.Response, error) {
 	}
 
 	createdAt := broker.CreatedAt
+
+	changes, err := query.LabelChangesFromJSON(r.Body)
+	if err != nil {
+		return nil, util.HandleLabelChangeError(err)
+	}
+	if r.Body, err = sjson.DeleteBytes(r.Body, "labels"); err != nil {
```
[the remark this answers is not in the document]
Comment (KirilKabakchiev) [reply api/broker/broker_controller.go]: Label changes are sifferent format from the actual labels and they are processed separatly

File: storage/postgres/broker.go [hunk 2]
```diff
@@ -31,40 +37,121 @@ type brokerStorage struct {
...
 	b.FromDTO(broker)
-	return create(ctx, bs.db, brokerTable, b)
+	id, err := create(ctx, bs.db, brokerTable, b)
+	if err != nil {
+		return "", err
+	}
+	return id, bs.createLabels(ctx, id, broker.Labels)
+}
+
+func (bs *brokerStorage) createLabels(ctx context.Context, brokerID string, labels types.Labels) error {
```
[the remark this answers is not in the document]
Comment (KirilKabakchiev) [reply storage/postgres/broker.go]: What is a label converter

File: storage/postgres/broker.go [hunk 3]
```diff
@@ -31,40 +37,121 @@ type brokerStorage struct {
...
 func (bs *brokerStorage) List(ctx context.Context, criteria ...query.Criterion) ([]*types.Broker, error) {
-	var brokerDTOs []Broker
-	if err := validateFieldQueryParams(Broker{}, criteria); err != nil {
+	rows, err := listWithLabelsByCriteria(ctx, bs.db, Broker{}, &BrokerLabel{}, brokerTable, criteria)
+	defer func() {
+		if rows == nil {
+			return
+		}
+		if err := rows.Close(); err != nil {
+			log.C(ctx).Errorf("Could not release connection when checking database s. Error: %s", err)
```
[the remark this answers is not in the document]
Comment (KirilKabakchiev) [reply storage/postgres/broker.go]: Typo

File: storage/postgres/broker.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dpanayotov) [reply storage/postgres/broker.go]: Names, interfaces and implementation not as expected, but something like
```golang
type LabelsConverter interface {
	FromDTO(entityID string, labels types.Labels) (error, ValidateableItems)
}

type ValidateableItems interface {
	Validate() error
	Items() []interface{}
}

func (bls *brokerLabels) Items() []interface{} {
	result := make([]interface{}, len(*bls))
	for i, label := range *bls {
		result[i] = label
	}
	return result
}
```

**Pull request merged**